core/src/data/value/binary_op/decimal.rs

```diff
@@ -0,0 +1,346 @@
+use {
+    super::TryBinaryOperator,
+    crate::{
+        data::{NumericBinaryOperator, ValueError},
+        prelude::Value,
+        result::{Error, Result},
+    },
+    rust_decimal::prelude::Decimal,
+    std::cmp::Ordering,
+    Value::*,
+};
+
+impl PartialEq<Value> for Decimal {
+    fn eq(&self, other: &Value) -> bool {
+        match other {
+            I8(other) => *self == Decimal::from(*other),
+            I64(other) => *self == Decimal::from(*other),
+            F64(other) => match Decimal::from_f64_retain(*other) {
+                Some(x) => *self == x,
+                _ => false,
```

Comment:
```suggestion
            F64(other) => Decimal::from_f64_retain(*other)
                .map(|x| *self == x)
                .unwrap_or(false),
```

How about this code?

Reply:
Looks good to me!  I'll also try to do something similar with other calls to Decimal::from_f64_retain...